.github/workflows/python-CI.yaml

```diff
@@ -17,12 +17,40 @@ jobs:
     runs-on: oss-4-core-runner
     steps:
       - uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          working-directory: python
+          list-files: json
+          filters: |
+            diff:
+              - "python/**/*.py"
+              - "python/**/*.toml"
+      - run: >-
+          echo "packages=$(echo '${{ steps.filter.outputs.diff_files }}'
+          | jq -r '.[]'
+          | sed -n
+          -e 's/^python\/instrumentation\/openinference-instrumentation-\([^/]*\)\/.*/\1/p'
+          -e 's/^python\/openinference-\(instrumentation\)\/.*/\1/p'
+          -e 's/^python\/openinference-\(semantic-conventions\)\/.*/\1/p'
+          | sort -u
+          | sed -e 's/semantic-conventions/semconv/' -e 's/-/_/' -e 's/^/\\b/'
+          | paste -sd "|" -)"
+          >> "$GITHUB_OUTPUT"
+        if: ${{ steps.filter.outputs.diff == 'true' }}
+        id: changes
```

Comment:
I do wonder if this logic feels a tad non deterministic. Github has ways for triggering actions on sub-directories. While it would be more work to setup CI then could report specific instrumentation CI failures rather than trying to do everything?

Reply:
that makes sense. we can do that as a follow-up